crates/hotshot/src/traits/networking/web_server_network.rs

```diff
@@ -962,58 +998,89 @@ impl<TYPES: NodeType + 'static> ConnectedNetwork<Message<TYPES>, TYPES::Signatur
                     .await;
             }
             ConsensusIntentEvent::PollForLatestQuorumProposal => {
-                let mut proposal_task = self.inner.latest_quorum_proposal_task.write().await;
-                if proposal_task.is_none() {
-                    // create new task
+                // Only start this task if we haven't already started it.
+                let mut cancel_handle = self.inner.cancel_latest_quorum_proposal_task.write().await;
+                if cancel_handle.is_none() {
+                    let inner = self.inner.clone();
+
+                    // Create sender and receiver for cancelling parent task
                     let (sender, receiver) = unbounded();
-                    *proposal_task = Some(sender);
+                    *cancel_handle = Some(sender);
+
+                    // Create the new task
+                    async_spawn(async move {
+                        loop {
```

Comment:
Do we need this loop here or can we just not return from `poll_webserver()` since it's already in a loop there, and uses the `running` atomic bool to do shutdown

Reply:
I think the `poll_webserver` loop returns once it gets the data needed

`                            MessagePurpose::Proposal => {
                                // Only pushing the first proposal since we will soon only be allowing 1 proposal per view
                                self.broadcast_poll_queue
                                    .write()
                                    .await
                                    .push(deserialized_messages[0].clone());

                                return Ok(());`

Reply:
Oh yeah, it doesn't in our case actually. I can do this

Reply:
Actually , some of the message types return on completion and some don't. What should we go with?

Reply:
I say don't return from poll for the `Latest_` message purpose.  Follow the pattern of handling votes